sklearn/decomposition/tests/test_incremental_pca.py

```diff
@@ -233,6 +234,27 @@ def test_incremental_pca_batch_signs():
         assert_almost_equal(np.sign(i), np.sign(j), decimal=6)
 
 
+def test_incremental_pca_partial_fit_small_batch():
+    # Test that there is no minimum batch size after the first partial_fit
+    # Non-regression test
+    rng = np.random.RandomState(1999)
+    n, p = 50, 3
+    X = rng.randn(n, p)  # spherical data
+    X[:, 1] *= 0.00001  # make middle component relatively small
+    X += [5, 4, 3]  # make a large mean
+
+    n_components = p
+    pipca = IncrementalPCA(n_components=n_components)
+    pipca.partial_fit(X[:n_components])
+    for idx in range(n_components, n):
+        pipca.partial_fit(X[idx : idx + 1])
+
+    pca = PCA(n_components=n_components)
+    pca.fit(X)
+
+    assert_almost_equal(pca.components_, pipca.components_, decimal=3)
```

Comment:
Nowadays, we rather use `assert_allclose` instead, which gives more control to set both relative and absolute tolerance thresholds. I would rather avoid introducing new calls to `assert_almost_equal` in new functions.

```suggestion
    assert_allclose(pca.components_, pipca.components_, atol=1e-3)
```